### PR TITLE
Enhancing Engine syntax

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|#17980|

The proposed changes aim to improve the readability and functionality of the decoders. These changes include:

- Replacing 'Source' with 'Parent' to refer to the originating decoder.
- Revising the helper function to resemble a standard programming language format.
- Introducing the usage of the '$' symbol to reference event fields during the check stage.
- Variables, implemented as fields inside the event and deleted from the event at the end of the asset processing.
- Custom fields no longer start with an anchor character, any field not present in the schema is a custom field.
- Added helper to erase custom fields, intended to use on the wazuh-core output so index is only populated with schema fields.

### CODE REVIEW

File: src/engine/test/source/builder/helperParser_test.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    else
+    {
+        ASSERT_TRUE(std::holds_alternative<base::Error>(result));
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    Builder,
+    HelperParserTest,
```
Comment (LucioDonda) [inline src/engine/test/source/builder/helperParser_test.cpp:32]: I would add some cases in order to clear out the parenthesis behavior:
```
HelperParserT(true, "test(arg1, ( arg2)", {.name = "test", .args = {"arg1", "( arg2"}}),  // Testing escaped parenthesis
HelperParserT(true, "test(arg1, ) arg2)", {.name = "test", .args = {"arg1", ") arg2"}}),  // Testing escaped parenthesis
HelperParserT(true, "test(arg1, ) arg2))))", {.name = "test", .args = {"arg1", ") arg2)))"}}),  // Testing escaped parenthesis
```

File: src/engine/source/builder/helperParser.hpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    HelperToken result;
+
+    if (re2::RE2::FullMatch(text, pattern, &result.name, &strArgs))
+    {
+        if (!strArgs.empty())
+        {
+            size_t pos = 0;
+            while ((pos = strArgs.find(',', pos)) != std::string::npos)
+            {
+                // if comma is the comma is scaped, skip it
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/builder/helperParser.hpp:77]: You're right

File: src/engine/source/logicExpression/src/logicExpressionParser.cpp [hunk 3]
```diff
@@ -19,10 +19,12 @@ namespace
...
-        base::utils::string::splitMulti(rawExpression,
-                                  base::utils::string::Delimeter(' ', false),
-                                  base::utils::string::Delimeter('(', true),
-                                  base::utils::string::Delimeter(')', true));
+        base::utils::string::splitMulti(rawExpression, base::utils::string::Delimeter(' ', false));
+
+    // TODO: Right now parenthesis conflict with helper builder syntax, so we force
+    // them to be separated by blank spaces. This should be fixed in the future parsing implementation.
+    //   base::utils::string::Delimeter('(', true),
+    //   base::utils::string::Delimeter(')', true));
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/logicExpression/src/logicExpressionParser.cpp:27]: It is a way of remembering the change to be made, for now we will leave it commented and we will eliminate it in the helpers parser issue

File: src/engine/source/logpar/src/logpar.cpp [hunk 4]
```diff
@@ -451,7 +425,7 @@ Logpar::Hlp Logpar::buildFieldParser(const parser::Field& field, const std::vect
     // Build target field
     // Special case <~> -> If name is ~, ignore output
     // Otherwise, set targetField
-    if (field.name.value != std::string {syntax::EXPR_CUSTOM_FIELD})
+    if (field.name.value != std::string {syntax::EXPR_WILDCARD})
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/logpar/src/logpar.cpp:428]: Yes, but they are part of the name, there are no differences between fields that start with ~ and those that do not, which are not in the ECS either. If the functionality of not mapping if the field is called `~` was preserved

File: src/engine/test/source/builder/helperParser_test.cpp (hunk 1 above)
Comment (juliancnn) [reply src/engine/test/source/builder/helperParser_test.cpp:32]: I agree

**Review by nbertoldo — APPROVED** [review]
LGTM!

**Review by nmkoremblum — CHANGES_REQUESTED** [review]

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**